src/services/account-states/account-state-engine.ts

```diff
@@ -1,118 +1,189 @@
-import { interventionsConfig, userLedActionsConfig } from './config';
+import { transitionConfiguration } from './config';
+import { StateDetails } from '../../data-types/interfaces';
 import { EventsEnum, MetricNames } from '../../data-types/constants';
-import { buildPartialUpdateAccountStateCommand } from '../../commons/build-partial-update-state-command';
-import { logAndPublishMetric } from '../../commons/metrics';
 import { StateTransitionError } from '../../data-types/errors';
+import logger from '../../commons/logger';
+import { logAndPublishMetric } from '../../commons/metrics';
+import { buildPartialUpdateAccountStateCommand } from '../../commons/build-partial-update-state-command';
 
-/**
- * State Engine Class
- * It stores the configuration for interventions and user actions transitions
- */
 export class AccountStateEngine {
-  private static readonly interventionConfigurations: InterventionTransitionConfigurations = interventionsConfig;
-  private static readonly userLedActionConfiguration: UserLedActionTransitionConfigurations = userLedActionsConfig;
+  private static readonly configuration = transitionConfiguration;
+  private static instance: AccountStateEngine;
+
+  private constructor() {}
+  public static getInstance() {
+    if (!AccountStateEngine.instance) {
+      AccountStateEngine.validateConfiguration();
+      AccountStateEngine.instance = new AccountStateEngine();
+    }
+    return AccountStateEngine.instance;
+  }
 
   /**
-   * Private helper method: given an account state object, it searches for that state in the intervention
-   * configuration object and returns the corresponding key
-   * @param accountState - object = blocked: Boolean; suspended: Boolean; resetPassword: Boolean; resetIdentity: Boolean
+   * Private method to validate configuration object in use by this class
    */
-  private static getInterventionEnumFromStateDetail(accountState: StateDetails) {
-    for (const key of Object.keys(AccountStateEngine.interventionConfigurations)) {
-      const state = (AccountStateEngine.interventionConfigurations[key] as InterventionEventDetails).state;
-      if (
-        accountState.blocked === state?.blocked &&
-        accountState.suspended === state?.suspended &&
-        accountState.reproveIdentity === state?.reproveIdentity &&
-        accountState.resetPassword === state?.resetPassword
-      )
-        return key as EventsEnum;
+  private static validateConfiguration() {
+    const accountStates = Object.keys(AccountStateEngine.configuration.nodes).sort(compareStrings);
+    const adjacencyLists = Object.keys(AccountStateEngine.configuration.adjacency).sort(compareStrings);
+    if (JSON.stringify(adjacencyLists) !== JSON.stringify(accountStates))
+      throw buildError(
+        MetricNames.INVALID_STATE_ENGINE_CONFIGURATION,
+        'Invalid state engine configuration detected. Adjacency mismatch',
+      );
+    for (const element of Object.values(AccountStateEngine.configuration.edges)) {
```

Comment:
```suggestion
    for (const edge of Object.values(AccountStateEngine.configuration.edges)) {
```
Seems slightly better to me